src/core/backbone_router/bbr_manager.cpp

```diff
@@ -505,21 +505,34 @@ NdProxyTable &Manager::GetNdProxyTable(void)
     return mNdProxyTable;
 }
 
-bool Manager::ShouldForwardDuaToBackbone(const Ip6::Address &aAddress)
+bool Manager::ShouldForwardDuaToBackbone(const Ip6::Address &srcAddress, const Ip6::Address &dstAddress)
 {
     bool              forwardToBackbone = false;
     Mac::ShortAddress rloc16;
     Error             error;
+    Child *           child;
+    bool              isMedChild = false;
+    Mac::Address      rloc;
 
     VerifyOrExit(Get<Local>().IsPrimary());
-    VerifyOrExit(Get<Leader>().IsDomainUnicast(aAddress));
+    VerifyOrExit(Get<Leader>().IsDomainUnicast(srcAddress));
+    VerifyOrExit(Get<Leader>().IsDomainUnicast(dstAddress));
 
-    VerifyOrExit(!mNdProxyTable.IsRegistered(aAddress.GetIid()));
+    VerifyOrExit(!mNdProxyTable.IsRegistered(dstAddress.GetIid()));
 
-    error = Get<AddressResolver>().Resolve(aAddress, rloc16, /* aAllowAddressQuery */ false);
+    // Perform address queries on behalf of minimal end device children
+    child      = Get<ChildTable>().FindChild(srcAddress);
+    isMedChild = (child != nullptr && child->GetDeviceMode().IsMinimalEndDevice());
+    error      = Get<AddressResolver>().Resolve(dstAddress, rloc16, isMedChild);
+
+    // If we successfully find an RLOC in our address cache, only attempt to forward
+    // to backbone if RLOC is not our own
     VerifyOrExit(error != kErrorNone || rloc16 == Get<Mle::MleRouter>().GetRloc16());
 
-    // TODO: check if the DUA is an address of any Child?
+    // If RLOC is our child, do not forward to backbone
+    rloc.SetShort(rloc16);
+    VerifyOrExit((Get<ChildTable>().FindChild(rloc, Child::kInStateValidOrRestoring)) == nullptr);
+
```

Comment:
We didn't check the DUA destination in Children because we were assuming that Children's DUAs should have been registered successfully so that we should never reach here (should exited at `VerifyOrExit(!mNdProxyTable.IsRegistered(dstAddress.GetIid()));`).

However, I think it's still helpful to check the Children's DUAs because it might be an 1.1 device with a manually configured DUA. That being said, I think we should use similar code in `MeshForwarder::UpdateIp6RouteFtd` to check if the DUA belongs to a Child using `Get<NeighborTable>().FindNeighbor`, before calling `Get<AddressResolver>().Resolve`. 

I think `Get<AddressResolver>().Resolve` never gets an RLOC16 of an MED child, anyway, because AddressResolver should never store RLOC16 of its MED Child since all Children IPv6 addresses and RLOC16s are already stored in `ChildTable`.